packages/block-editor/src/components/rich-text/format-toolbar/style.scss

```diff
@@ -1,7 +1,24 @@
-.block-editor-format-toolbar .components-dropdown-menu__toggle {
-	width: 32px;
+.block-editor-format-toolbar {
+	// Size the formatting buttons to be optically balanced.
+	// Specifically, buttons in a toolbar group are not all 48px.
+	// They are 36px, except for the first and the last, which have
+	// an additional 6px padding left or right.
+	.components-toolbar > div {
+		width: 48px - 6px - 6px;
```

Comment:
I wonder if it's possible for third-party plugins to add formatting buttons that are not icons but text (so not square).
Can we achieve the same behavior without forcing a width?

Reply:
Addressed this in https://github.com/WordPress/gutenberg/pull/19344/commits/0417c927125c0b6a4f36d9ce93093c593a8c2568, but it's a little messy and specific. Advice is welcome, but I'm also hoping to revisit this.